monai/data/image_reader.py

```diff
@@ -1015,14 +1043,34 @@ def _get_spatial_shape(self, img):
         spatial_rank = max(min(ndim, 3), 1)
         return np.asarray(size[:spatial_rank])
 
-    def _get_array_data(self, img):
+    def _get_array_data(self, img, filename):
         """
         Get the raw array data of the image, converted to Numpy array.
 
         Args:
             img: a Nibabel image object loaded from an image file.
 
         """
+        if self.to_gpu:
+            file_size = os.path.getsize(filename)
+            image = cp.empty(file_size, dtype=cp.uint8)
```

Comment:
Will it be an issue when data dtype is not uint8?

Reply:
Hi @KumoLiu , thanks for the question.
CUPY uses uint8 for handling bytes data. Since the loaded data is bytes, cp.uint8 is used here.